python_modules/libraries/dagster-airlift/dagster_airlift/core/monitoring_job/builder.py

```diff
@@ -1,76 +1,144 @@
+    SensorEvaluationContext,
 )
 from dagster._core.execution.context.op_execution_context import OpExecutionContext
-from dagster._core.storage.dagster_run import RunsFilter
+from dagster._core.storage.dagster_run import DagsterRun, RunsFilter
 from dagster._grpc.client import DEFAULT_SENSOR_GRPC_TIMEOUT
-from dagster._time import datetime_from_timestamp, get_current_datetime
+from dagster._record import record
+from dagster._serdes import deserialize_value, serialize_value
+from dagster._time import get_current_datetime
 from dagster_airlift.core.airflow_defs_data import AirflowDefinitionsData
 from dagster_airlift.core.airflow_instance import AirflowInstance
 from dagster_airlift.core.monitoring_job.event_stream import persist_events
-from dagster_airlift.core.monitoring_job.utils import (
-    augment_monitor_run_with_range_tags,
-    get_range_from_run_history,
-    structured_log,
-)
+from dagster_airlift.core.monitoring_job.utils import structured_log
 from dagster_airlift.core.utils import monitoring_job_name
+from dagster_shared.serdes import whitelist_for_serdes
 
 MAIN_LOOP_TIMEOUT_SECONDS = DEFAULT_SENSOR_GRPC_TIMEOUT - 20
 DEFAULT_AIRFLOW_SENSOR_INTERVAL_SECONDS = 30
 START_LOOKBACK_SECONDS = 60  # Lookback one minute in time for the initial setting of the cursor.
 
 
+@whitelist_for_serdes
+@record
+class AirflowMonitoringJobSensorCursor:
+    range_start_iso: str
+    range_end_iso: str
+
+    def to_config(self) -> "MonitoringConfig":
+        return MonitoringConfig(
+            range_start_iso=self.range_start_iso, range_end_iso=self.range_end_iso
+        )
+
+    def to_tags(self) -> dict[str, str]:
+        return {
+            "range_start_iso": self.range_start_iso,
+            "range_end_iso": self.range_end_iso,
+        }
+
+    def advance(self) -> "AirflowMonitoringJobSensorCursor":
+        return AirflowMonitoringJobSensorCursor(
+            range_start_iso=self.range_end_iso, range_end_iso=get_current_datetime().isoformat()
+        )
+
+
 # IMPROVEME BCOR-102: We should be able to replace the sensor from the original Airlift functionality with this job.
 @beta
 def build_airflow_monitoring_defs(
     *,
     airflow_instance: AirflowInstance,
 ) -> Definitions:
     """The constructed job polls the Airflow instance for activity, and inserts asset events into Dagster's event log."""
+    return Definitions(
+        jobs=[build_monitoring_job(airflow_instance=airflow_instance)],
+        sensors=[build_monitoring_sensor(airflow_instance=airflow_instance)],
+    )
 
-    @job(name=monitoring_job_name(airflow_instance.name))
-    def airflow_monitoring_job():
-        _build_monitoring_op(airflow_instance)()
 
-    @schedule(
-        job=airflow_monitoring_job,
-        cron_schedule="* * * * *",
-        name=f"{airflow_instance.name}__airflow_monitoring_job_schedule",
-        default_status=DefaultScheduleStatus.RUNNING,
+class MonitoringConfig(Config):
+    range_start_iso: str
+    range_end_iso: str
```

Comment:
I think these should just be called range_start and range_end -- we can add a description with pydantic.Field if we want to make it clear what the expected format is when someone's editing this in the launchpad.

The properties can just be `range_start_datetime` and `range_end_datetime`, or we could even just have a `range_time_window` which returns a TimeWindow.